### PR TITLE
added git-push in docs

### LINKED ISSUE #452: no `git push` in the docs
I have not found any `git push` command in Get Started and in other places. I understand the motivation behind - users should know how to work with Git and we don't want to spend time to set up Git remote in the docs.

However, some user might be confused. They might not know Git enough. Or they might decide that `dvc push` executes `git push`.

It might worth clarifying this in the docs.

### PR DESCRIPTION
**Pull request by Naba7**
Fix #452

### CODE REVIEW

File: static/docs/commands-reference/push.md [hunk 1]
```diff
@@ -24,7 +24,9 @@ environments and preserving data versions (input datasets, intermediate results,
 models, metrics, etc) remotely (S3, SSH, GCS, etc) are the most common use cases
 for these commands.
 
-The `dvc push` command allows one to upload data to remote storage.
+The `dvc push` command allows one to upload data to remote storage. It is run
+along with 'git commit' and
```
Comment (shcheklein) [inline static/docs/commands-reference/push.md]: ' change to `

File: static/docs/commands-reference/push.md [hunk 2]
```diff
@@ -24,7 +24,9 @@ environments and preserving data versions (input datasets, intermediate results,
...
+[git push](https://git-scm.com/docs/git-push/1.5.5.2).
```
Comment (shcheklein) [inline static/docs/commands-reference/push.md]: let's remove the link, it looks weird that we are linking only git push. And it should be clear to ppl what does git push mean

File: static/docs/commands-reference/push.md [hunk 3]
```diff
@@ -24,7 +24,9 @@ environments and preserving data versions (input datasets, intermediate results,
 models, metrics, etc) remotely (S3, SSH, GCS, etc) are the most common use cases
 for these commands.
 
-The `dvc push` command allows one to upload data to remote storage.
+The `dvc push` command allows one to upload data to remote storage. It is run
```
Comment (shcheklein) [inline static/docs/commands-reference/push.md]: ... usually ...

File: static/docs/get-started/share-data.md [hunk 4]
```diff
@@ -12,7 +12,9 @@ The same way as with Git remote, it ensures that your data files and your models
 are safely stored remotely and are shareable. It means that this data could be
 pulled by your team or you when you need it.
 
-Usually, you run it along with `git commit` and `git push` to save changed
+Usually, you run it along with `git commit` and
+[git push](https://git-scm.com/docs/git-push/1.5.5.2)
```
Comment (shcheklein) [inline static/docs/get-started/share-data.md]: no need in this link, tbh

File: static/docs/commands-reference/push.md (hunk 2 above)
Comment (Naba7) [reply static/docs/commands-reference/push.md]: okay! Shall i explain ``git push`` instead?

File: static/docs/get-started/share-data.md (hunk 4 above)
Comment (Naba7) [reply static/docs/get-started/share-data.md]: Shall I describe it here ? IMO there is no need to explain ``git push``

Comment (shcheklein) [reply static/docs/get-started/share-data.md]: I would change the paragraph, to emphasize that `dvc push` does not include or run `git push` and takes care only of data, that you still need to remember to run `git commit` and `git push`

Comment (Naba7) [reply static/docs/get-started/share-data.md]: okay.

File: static/docs/commands-reference/push.md [hunk 5]
```diff
@@ -24,9 +24,10 @@ environments and preserving data versions (input datasets, intermediate results,
 models, metrics, etc) remotely (S3, SSH, GCS, etc) are the most common use cases
 for these commands.
 
-The `dvc push` command allows one to upload data to remote storage. It is run
-along with 'git commit' and
-[git push](https://git-scm.com/docs/git-push/1.5.5.2).
+The `dvc push` command allows one to upload data to remote storage. It is 
+usually run along with `git commit` and `git push`. `dvc push` and `git push`
```
Comment (shcheklein) [inline static/docs/commands-reference/push.md]: intention is not to explain the command (git push), but to emphasize that `dvc push` is not doing anything magical, that it's not running git push for you or something. It's already done by the first two sentences to some extent. I would just modify them a little bit to be more specific:

> The `dvc push` command allows one to upload data to remote storage. It does not save changes to code or DVC-files. You still need to run `git commit` and `git push` to save them.

Something like this ^^

@jorgeorpinel could you take a look, does the phrase above look reasonable?

Comment (Naba7) [reply static/docs/commands-reference/push.md]:
```
The `dvc push` command allows one to upload data to remote storage. It doesn't
update any modified changes in the code or DVC-files. The changes made in the 
the files or addition of any new file still needs to be saved by using        
`git commit` and `git push`.
```
Is it good?

Comment (shcheklein) [reply static/docs/commands-reference/push.md]: `update any modified change` -> save any changes

"The changes made in the the files or addition of any new file still" -> Those should be saved ...

File: static/docs/get-started/share-data.md [hunk 6]
```diff
@@ -15,6 +15,10 @@ pulled by your team or you when you need it.
 Usually, you run it along with `git commit` and `git push` to save changed
 [DVC-files](/doc/user-guide/dvc-file-format) to Git.
 
+The `dvc push` command allows one to upload data to remote storage. It doesn't
+save any changes in the code or DVC-files. Those should be saved by using      
+`git commit` and `git push`.
+
 See `dvc push` for more details and options for this command.
```
Comment (jorgeorpinel) [inline static/docs/get-started/share-data.md]: Sorry, I was a little distracted last week. I do like the paragraph you guys came up with, but I will remove the paragraph separation above in my active PR for this doc. Thanks

**Pull request merged**